### DISCUSSION TITLE
Shadcn and Getx

### DISCUSSION THREAD

**Discussion by Khalifares** [nank1ro/flutter-shadcn-ui#63] [Q&A] [answered]
Hi there!

First of all I'd like to say that I'm learning a lot by exploring your code so a big thank you!

I have a question:
Is it possible to use Getx pre-configured widget GetMaterialApp in combination with ShadApp? If so, how?

Thanks in advance,

**Comment by nank1ro** [comment]
Hi, I am happy that you are learning new things.

Please, don't use GetX if you can, especially if you are a beginner.
I did the same mistake when I started, and I had to learn everything from scratch after changing state management.
Why GetX is bad:
- getx makes you learn getx, not flutter
- getx is a very large library but has very little documentation
- getx makes you avoid learning a crucial part of flutter (the BuildContext)
- getx makes you learn bad practices and uses bad practices in its own code (GlobalKeys are overused)
- getx makes your code very dependent on getx, making it extremely hard to replace later
- getx does not actually offer any unique features
- getx has a high like count on pub.dev but we suspect it to be acquired through abnormal means

More opinions from notable persons:
https://twitter.com/scottstoll2017/status/1468431032240246784?s=20
https://twitter.com/filiphracek/status/1468565501701939203

You can choose any other state management solution that provides a solid architecture and has solid bases.
If I have to suggest for some I would say:
- [solidart](https://docs.page/nank1ro/solidart) I'm the creator, I created it because I loved the simplicity of SolidJS in the JS world and wanted to have the same in Dart. Respects the best practices of Flutter.
- [provider](https://pub.dev/packages/provider) Created by Remi, the creator of Riverpod also. Riverpod is newer but not beginner friendly IMHO.

**Reply by LaBolaCuba** [reply]
So, how to mix getx and ShadApp ?, is this possible ?

**Reply by nank1ro** [reply, marked as answer]
From the v0.13.3 yes.
https://flutter-shadcn-ui.mariuti.com/#shadcn--custom-app
